### DISCUSSION TITLE
Motor go wrong way

### DISCUSSION THREAD

**Discussion by posssson** [doudar/SmartSpin2k#397] [Q&A] [answered]
Hi, I didn't use it since 5-6 months and I updated the code and now the motor goes in the wrong way.. For exemple, the targer is 200 watts and he go to 0 watts and so stop, and when I change the param of motor direction he go to 600w and try to go more... I tried to change all params, change the motor, all wires... Any ideas?

**Comment by doudar** [comment, marked as answer]
The easiest is to change the motor direction in the settings. Also in order to have the most current html files, it will need to be connected to a 2.4ghz WiFi network

**Reply by posssson** [reply]
I did to change the direction and I have the same problem and I am connected to a 2.4 WIFI :(

**Reply by posssson** [reply]
I did a clean erase of the esp and take the last develop commit

**Reply by posssson** [reply]
I realy don't understand because the sign of target incline change but the motor turn always in the same direction... What can be wrong?

**Reply by eMadman** [reply]
When you use the web shifter (no app connected, just click the shifter), does shifting up and down turn the motor in the correct direction?

What app are you using for training? Does this occur in other apps?  You can test with MyWhoosh since it's pretty feature-rich and completely free.  I've verified it works with SS2K and use it for a lot of my own testing.

**Reply by posssson** [reply]
Same problem with web shiffter both go same way...I use zswift and flux

**Reply by doudar** [reply]
Is this a hand built unit not using the pcb? If so check the dir pin connection.

**Reply by posssson** [reply]
So you was right. I use PCB build so I thought wires was ok but indeed the connexion of the dir pin was broken after only 1 year of use. I could'nt fix it behind esp and TMC so I make wire directly above PCB. Thank you !!!!

**Reply by doudar** [reply]
Thanks for the feedback!
